staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -172,7 +183,8 @@ func (r *crdHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		r.delegate.ServeHTTP(w, req)
 		return
 	}
-	if !apiextensions.IsCRDConditionTrue(crd, apiextensions.Established) {
+	if !apiextensions.IsCRDConditionTrue(crd, apiextensions.NamesAccepted) &&
```

Comment:
would add a comment to point out that with this there is a small chance that a CRD is being served (because NamesAccepted=true) and then "unserved" because another names updates lead to a conflict, but the establising controller was not fast enough to set Established=true. But this danger is small and self-healing, so we accept it.